sgv2-restapi/pom.xml

```diff
@@ -262,6 +262,33 @@
       <plugin>
         <artifactId>maven-surefire-plugin</artifactId>
       </plugin>
+      <!-- Plugin for building Docker images -->
+      <plugin>
+        <groupId>com.google.cloud.tools</groupId>
+        <artifactId>jib-maven-plugin</artifactId>
+        <version>3.1.4</version>
+        <configuration>
+          <from>
+            <image>adoptopenjdk:8</image>
+          </from>
+          <to>
+            <image>docker.io/stargateio/stargate-restapi:${project.version}</image>
+          </to>
+          <!-- See: https://github.com/GoogleContainerTools/jib/tree/master/jib-maven-plugin#container-object-->
+          <container>
+            <creationTime>USE_CURRENT_TIMESTAMP</creationTime>
+            <filesModificationTime>${maven.build.timestamp}</filesModificationTime>
+            <ports>
+              <port>8082</port>
+            </ports>
+            <jvmFlags>
+              <jvmFlag>-Ddw.stargate.grpc.host=localhost</jvmFlag>
```

Comment:
Hmm not sure what the defaults mean here wrt to the image building.. Shouldn't these defaults be in the default configuration and not bounded to the images itself? Furthermore, will environment variables for the same be treated as higher priority and overwrite those?

Reply:
The properties can be overridden using the `JAVA_TOOL_OPTIONS` environment variable as described here https://github.com/GoogleContainerTools/jib/blob/master/docs/faq.md#how-do-i-set-parameters-for-my-image-at-runtime. 

I agree there is some awkwardness with using Java properties. @tatu-at-datastax we could also discuss changing the REST API service so that it uses environment variables in addition to / instead of Java properties? Whichever approach we choose, I think it's good practice to have the default settings defined in the image. We should also document how to override them in the README. After we decide how the properties will be set I'd be happy to update this PR to build the image and update the documentation accordingly.

Reply:
Well I guess that Dropwizard can accept properties coming from different sources right?

Reply:
Yes, but I believe the plan is to remove Dropwizard imminently

Reply:
Yes but other frameworks do the same, env vars always have highest priority..

Reply:
If I remember correctly, it is possible to pipe in ENV variables to DropWizard config, but it is not enabled by default (for security reasons; not everyone wants to allow such overrides, I think).
If env variables are the main mechanism (k8s seems to favor them; Java world considered them ancient and obsolete, go figure -- everything old is now new hotness :) ), we could also just consider passing them through system properties (`-Ddw.some.setting=${MY_ENV_VARIABLE)`) for now? And then consider rework when we re-platform.

Reply:
> I think it's good practice to have the default settings defined in the image

Typically defaults are defined in code that way you have the same behavior inside and outside of Docker. I am +1 on environment variables though.

Reply:
pushed a change that uses environment variables to set java properties, as suggested by @tatu-at-datastax. While this does work, it is slightly awkward and does require setting defaults for all environment variables.